### PR TITLE
CI: Use custom action, activate linter

### LINKED ISSUE #90: [CD] Make 'tests.yml' workflow more DRY
Currently the following lines are repetitive. It would be nice to make them follow the Do-Not-repeat-Yourself principle by separating them in another action/workflow.
```yaml
    - uses: actions/checkout@v4
    - name: Install and configure poetry
      run: |
        pipx install poetry
        poetry config virtualenvs.in-project true
    - uses: actions/setup-python@v5
      with:
        python-version: ${{ env.PYTHON_VERSION }}
        cache: 'poetry'
    - name: Setup poetry
      run: poetry install
```

### LINKED ISSUE #92: [CD] Speed up github actions
- [x] Cache Poetry itself (potential [solution](https://github.com/pypa/pipx/discussions/1051#discussioncomment-6946826))
 - [x] Replace `npm` with `pnpm` or other fast alternative
 - [x] Cache node packet manager itself
 - [x] Apply other optimizations

### PR DESCRIPTION
**Pull request by Privat33r-dev**
Fixes #90 
Fixes #92

### CODE REVIEW

**Comment by Privat33r-dev** [comment]
Actually I wanted to make this PR on my clairvoyance repo for testing, but it's ok, I guess :)

**Comment by Privat33r-dev** [comment]
Rearranging imports in `oracle.py` is kind of cosmetic change and it's the only change that is required to pass the `isort` test, so I feel that it would be within the scope of this PR.

**Comment by Privat33r-dev** [comment]
Also I don't want this composed action to be on my account since I can't predict what will happen years later, so I want it to be on @nikitastupin or gh organization account. I will mark it as "ready for release" it once we'll resolve the situation around action.

Also action might be renamed, something boring like `cached-poetry-install-action`, I used obscured name as a continuation of `clairvoyance` naming tradition :)

File: .github/actions/poetry-setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    description: "Version of Poetry to install"
+    default: "1.8.2"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Pipx Local Variables
+      id: pipx-env-setup
+      run: |
+        SEP="${{ !startsWith(runner.os, 'windows') && '/' || '\\' }}"
```
Comment (nikitastupin) [inline .github/actions/poetry-setup/action.yml:18]:
```suggestion
        SEP="${{ startsWith(runner.os, 'windows-') && '\\' || '/' }}"
```

Comment (nikitastupin) [reply .github/actions/poetry-setup/action.yml:18]: Opinionated simplification :)

**Review by nikitastupin — APPROVED** [review]
LGTM!

**Pull request merged**